x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/fielddata/ScriptBooleanFieldData.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.fielddata;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.IndexFieldDataCache;
+import org.elasticsearch.index.fielddata.IndexNumericFieldData;
+import org.elasticsearch.index.fielddata.plain.LeafLongFieldData;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.search.aggregations.support.CoreValuesSourceType;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+import org.elasticsearch.xpack.runtimefields.BooleanScriptFieldScript;
+
+import java.io.IOException;
+
+public final class ScriptBooleanFieldData extends IndexNumericFieldData {
+
+    public static class Builder implements IndexFieldData.Builder {
+        private final String name;
+        private final BooleanScriptFieldScript.LeafFactory leafFactory;
+
+        public Builder(String name, BooleanScriptFieldScript.LeafFactory leafFactory) {
+            this.name = name;
+            this.leafFactory = leafFactory;
+        }
+
+        @Override
+        public ScriptBooleanFieldData build(IndexFieldDataCache cache, CircuitBreakerService breakerService, MapperService mapperService) {
+            return new ScriptBooleanFieldData(name, leafFactory);
+        }
+    }
+
+    private final String fieldName;
+    private final BooleanScriptFieldScript.LeafFactory leafFactory;
+
+    private ScriptBooleanFieldData(String fieldName, BooleanScriptFieldScript.LeafFactory leafFactory) {
+        this.fieldName = fieldName;
+        this.leafFactory = leafFactory;
+    }
+
+    @Override
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    @Override
+    public ValuesSourceType getValuesSourceType() {
+        return CoreValuesSourceType.BOOLEAN;
+    }
+
+    @Override
+    public ScriptBooleanLeafFieldData load(LeafReaderContext context) {
+        try {
+            return loadDirect(context);
+        } catch (Exception e) {
+            throw ExceptionsHelper.convertToElastic(e);
+        }
+    }
+
+    @Override
+    public ScriptBooleanLeafFieldData loadDirect(LeafReaderContext context) throws IOException {
+        return new ScriptBooleanLeafFieldData(new ScriptBooleanDocValues(leafFactory.newInstance(context)));
+    }
+
+    @Override
+    public NumericType getNumericType() {
+        return NumericType.DOUBLE;
```

Comment:
I never realized this either: booleans are double????

Reply:
Ooops, copy and paste error. Fixing.